src/providers/azure/azure.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The azure provider fetches a configuration from the Azure OVF DVD.
+
+package azure
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/src/log"
+	"github.com/coreos/ignition/src/providers"
+	"github.com/coreos/ignition/src/providers/util"
+)
+
+const (
+	name           = "azure"
+	initialBackoff = 1000 * time.Millisecond
+	maxBackoff     = 30 * time.Second
+	configDevice   = "/dev/sr0"
```

Comment:
It's probably safe enough, but I think we could be more correct with a bit of work in identifying the microsoft azure virtual device @ udev and creating a specifically named device entry symlink pointing at the /dev/srN it was identified on.